src/pages/NewProgram/views/ProgramFormProvider/ProgramFormProvider.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect } from "react";
+import { FormProvider } from "react-hook-form";
+import { useSnackbar } from "notistack";
+
+import { Button } from "@mui/material";
+import NavigateNextIcon from "@mui/icons-material/NavigateNext";
+
+import {
+  useNewProgramForm,
+  AddProgramFormFields,
+} from "../../../../hooks/formHooks/program/useNewProgramForm";
+import { useFormSteps } from "../../../../hooks/formHooks/formStepHook/useFormSteps";
+
+import {
+  FormWrapper,
+  FirstFormStepWrapper,
+  SecondFormStepWrapper,
+  ThirdFormStepWrapper,
+  FormActions,
+} from "./styles/ProgramForm.styles";
+import {
+  ProgramTitle,
+  ProgramLevel,
+  ProgramFrequency,
+  ProgramLength,
+  Program,
+  ProgramPrice,
+  ProgramDescription,
+  Image,
+} from "./views/ProgramForm/Program.form";
+import { ErrorMessage } from "../../../../components";
+import { Program as ProgramInterface } from "../../../../shared/interfaces";
+
+type ProgramFormProviderProps = {
+  editProgram?: ProgramInterface;
+};
+
+export const ProgramFormProvider: React.FC<ProgramFormProviderProps> = ({
+  editProgram,
+}) => {
+  const {
+    methods,
+    appendProgramField,
+    removeProgramField,
+    programFields,
+    canSubmit,
+    onSubmit,
+  } = useNewProgramForm({ editProgram });
+  const { currentStep, nextStep } = useFormSteps(!!editProgram);
+  const { enqueueSnackbar } = useSnackbar();
+
+  const {
+    watch,
+    trigger,
+    handleSubmit,
+    formState: { errors, isSubmitSuccessful },
+  } = methods;
+
+  const programLength = watch(AddProgramFormFields.PROGRAM_LENGTH);
+  const programFrequency = watch(AddProgramFormFields.FREQUENCY_PER_WEEK);
+
+  useEffect(() => {
+    if (isSubmitSuccessful) {
+      enqueueSnackbar("Program added successfuly.", {
```

Comment:
you can create a hook so that values such as autoHideDuration will be fixed for the entire application, and you will only provide status and message